modules/registry/registry.go

```diff
@@ -203,3 +253,55 @@ func Run(ctx context.Context, img string, opts ...testcontainers.ContainerCustom
 
 	return c, nil
 }
+
+// SetDockerAuthConfig sets the DOCKER_AUTH_CONFIG environment variable with
+// authentication for with the given host, username and password sets.
+// It returns a function to reset the environment back to the previous state.
+func SetDockerAuthConfig(host, username, password string, additional ...string) (func(), error) {
```

Comment:
I'm going to take this PR and add the docs so you don't have to deal with that burden. Will push commits on top of these ones